code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1445,3 +1446,47 @@ obj/machinery/computer/pandemic/proc/replicator_cooldown(var/waittime)
 		ui = new(user, src, ui_key, "chem_heater.tmpl", "ChemHeater", 350, 270)
 		ui.set_initial_data(data)
 		ui.open()
+
+///////////////////////////////////////////////////////////////////////////
+
+/obj/machinery/chem_dispenser/drinks
+	name = "soda dispenser"
+	anchored = 1
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "soda_dispenser"
+	energy = 100
+	max_energy = 100
+	amount = 30
+	recharge_delay = 5
+	uiname = "Soda Dispenser"
+	dispensable_reagents = list("water","ice","coffee","cream","tea","icetea","cola","spacemountainwind","dr_gibb","space_up","tonic","sodawater","lemon_lime","sugar","orangejuice","limejuice","tomatojuice")
+
+/obj/machinery/chem_dispenser/drinks/attackby(var/obj/item/O as obj, var/mob/user as mob)
+
+		if(default_unfasten_wrench(user, O))
+				return
+
+		if (istype(O,/obj/item/weapon/reagent_containers/glass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/drinkingglass) || \
+				istype(O,/obj/item/weapon/reagent_containers/food/drinks/shaker))
+
+				if (beaker)
+						return 1
+				else
+						src.beaker =  O
+						user.drop_item()
+						O.loc = src
+						update_icon()

```

Comment:
chem_dispenser and chem_dispenser/drinks do not have a proper update_icon() proc. This is the result of mindless copypasta, and no reviewers picked up on this... :see_no_evil: